src/main/java/com/openlattice/chronicle/services/ChronicleServiceImpl.java

```diff
@@ -333,26 +642,38 @@ public UUID registerDatasource(
             return null;
         }
 
-        UUID deviceEntityKeyId = reserveDeviceEntityKeyId( datasourceId, datasource, dataIntegrationApi, dataApi );
-        if (deviceEntityKeyId == null) {
-            logger.error("Unable to reserve deviceEntityKeyId, dataSourceId = {}, studyId = {}, participantId = {}", datasourceId, studyId, participantId);
-            return null;
```

Comment:
Why are you getting rid of the null check?

Reply:
Not sure why I did that, but I am reverting. The null check is important.